### PR TITLE
Nearest neighbour

### PR DESCRIPTION
**Pull request by lquenti**

### CODE REVIEW

**Comment by lquenti** [comment]
I manually verified the 5x5 by looking at the traces, the 10x10 is probably correct since the rust and python version cant be wrong if they were right for 5x5? :D

File: src/solvers/approximate/nearest_neighbour.rs [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+/// Call [`single_nearest_neighbour`] for every starting node.
+pub fn nearest_neighbour<const MODE: usize>(graph_matrix: &NAMatrix) -> Solution {
+    #[cfg(feature = "mpi")]
+    if MODE == MPI_COMPUTATION {
+        return nearest_neighbour_mpi(graph_matrix);
+    }
+    n_nearest_neighbour::<MODE>(graph_matrix, graph_matrix.dim())
+}
+
+// TODO refactor me into the structure
```
Comment (johann-cm) [inline src/solvers/approximate/nearest_neighbour.rs:108]: Explain a bit more, how you divide the work among the mpi processes.

**Comment by lquenti** [comment]
Feedback:
- [ ] ~~lower bound print (sth sth MPI finalize)~~
- [x] remove todo
- [ ] ~~Move MPI driver code into `lib.rs`~~

its not breaking with finalize as we do not have any lower bound MPI yet

**Comment by lquenti** [comment]
fuck me i forgot to improve push the mpidocs so I lost them. I wrote them now in #29

**Pull request merged**